diff --git a/scripts/xdg-mime.in b/scripts/xdg-mime.in
index e85fd62..8fb6866 100644
--- a/scripts/xdg-mime.in
+++ b/scripts/xdg-mime.in
@@ -241,7 +241,7 @@ make_default_generic()
     # Add $2=$1 to XDG_CONFIG_HOME/mimeapps.list
     xdg_config_home="$XDG_CONFIG_HOME"
     [ -n "$xdg_config_home" ] || xdg_config_home="$HOME/.config"
-    default_file="$xdg_config_home/mimeapps.list"
+    default_file=$(readlink -f -- $xdg_config_home/mimeapps.list)
     DEBUG 2 "make_default_generic $1 $2"
     DEBUG 1 "Updating $default_file"
     [ -f "$default_file" ] || touch "$default_file"
